tests/test_any.py

```diff
@@ -0,0 +1,45 @@
+import io
+import glob
+import os.path
+
+import pytest
+
+from readme_renderer.any import render
+
+
+@pytest.mark.parametrize(
+    ("src_filename", "html_filename"),
+    [
+        (fn, os.path.splitext(fn)[0] + ".html")
+        for fn in glob.glob(
+            os.path.join(os.path.dirname(__file__),
+                         "fixtures", "test_*.rst")
+        ) + glob.glob(
+            os.path.join(os.path.dirname(__file__),
+                         "fixtures", "markdown", "*.md")
+        )
+    ],
+)
+def test_any_fixtures(src_filename, html_filename):
+    too_short = {
+        'misc.md',
+        'test_rst_004.rst',
+        'test_rst_005.rst',
+        'test_rst_006.rst',
+        'test_rst_007.rst'}
+    if os.path.basename(src_filename) in too_short:
+        """sniff won't match in any cases for those files being too short
+        tests to be sniffed correctly (like just an "<a href", or just
+        a <script>alert("Hello")</script>) """
+        return
+    # Get our Markup
+    with io.open(src_filename, encoding='utf-8') as f:
+        markup = f.read()
```

Comment:
`open` from Python 2.7 had no encoding parameter, while we want this to work on pypi-legacy, we still need to keep Python 2 compatibility, also that's how tests are written for rst, so it's probably nice to write them the same way for consistency, and as I don't want to modify rst tests in a markdown branch, I think we should keep this that way. I'm not fluent in Python 2 / Python 3 compatible code, but as far as I searched it looks the right way to do it.